### ISSUE TITLE
One/two actions are done repeatedly hundreds of times (label added and removed)

### ISSUE THREAD

**Issue by 0xc0170** [Mergifyio/mergify-engine#3307]
## Expected Behavior

Action is done only once,  but it is being repeatedly done

## Actual Behavior

https://github.com/ARMmbed/mbed-os/pull/14960 - See the pull request, at the end, there are >3000 thousands actions done by mergify. One label being added/removed constantly. I noticed this behavior the last week, I thought it was a glitch with one pull request. As I saw it today again with another one, I started looking at the config and pull request itself, I can't find why this would be happening. We haven't changed our mergify config for 2 months. 

While it was happening, I verified Checks. There were 2 conditions (add label needs: work) satisfied. I can't figure out why needs: work was removed and added back and forth.

## Steps to Reproduce the Problem

-

## Specifications

  - Pull Request URL: https://github.com/ARMmbed/mbed-os/pull/14960
  - Mergify Config URL: https://github.com/ARMmbed/mbed-os/blob/master/.mergify.yml

**Comment by sileht** [comment]
One month ago, we fix a bug where labels weren't set correctly in some cases.

It looks like Mergify just do what the configuration said:

`Rule: label needs: work when Jenkins CI failed - pr head (label)` -> add the `needs: work` label

`Rule: remove release version and needs labels (label)` -> remove all labels

So my suggestion is to add to `Rule: label needs: work when Jenkins CI failed - pr head (label)` the condition `-closed`

So the pull request will not get the label added back with the pull request is closed as `Rule: remove release version and needs labels (label)`  remove all labels when a pull request is closed.

**Comment by 0xc0170** [comment]
I found another pull request where it was happening (it was opened at the time https://github.com/ARMmbed/mbed-os/pull/15080#event-5372744440). Shouldn't mergify check if an action was already done on the pull request and there was no change since the last time ? 

I created a patch to address closed PRs. Thanks for the suggestion.
